### DISCUSSION TITLE
Install to a mini PC without Linux install first?

### DISCUSSION THREAD

**Discussion by tf5bassist** [bitfocus/companion#2925] [Q&A]
At work, we have a few old Dell Wyse 5070 thin clients that make excellent mini PCs for low-power applications.  I've used one I've bought personally at home to replace my rpi4 for Home Assitant, for example.  I want to use one of these for Companion, but I'm unsure if it will work.

We're pretty strict on IT security, so my options are somewhat limited.  But I was hoping that I would be able to flash the CompanionPi image to a safe USB stick with Balena Etcher and then install it to the Wyse.  I need to check to see whether it has an SSD or eMMC storage, but we should be able to throw an SSD in if there isn't one.

This will live on our video production LAN, which is segregated from our corporate LAN and only has devices in our production desk and doesn't have access to the internet or the corporate LAN.  Because this isn't a company-managed device I can't put it on the corporate LAN to run the curl script in the documentation, so I think this is our only option.

So I guess my main questions are:

- If I flash the companion pi image to a USB drive and boot from that, will it allow me to install Companion to the storage in the Wyse terminal, or does it force you to run off of the boot media, like an rpi would technically run off of the mini SD card?

**Comment by Julusian** [comment]
The companionpi image is form arm64 systems, so no it wont run on that machine which google says has an intel (x64) cpu.

You could probably do the normal ubuntu/debian/whatever install, then copy across a zip downloaded from the website and run that, optionally performing a similar level of headless setup as the install script would do.
But I think there are at least a couple of system libraries that you would also need to install for it to run.

Is there not a guest network that it could be connected to while installing, or perhaps tether it to a phone?

**Reply by tf5bassist** [reply]
Ahh, thanks for the clarification.  I thought it was odd that some of the documentation insinuated that you could install the CompanionPi image to an x86 machine with some somewhat nebulous wording, so that makes sense.

Yeah, I think that installing the Linux distro is probably the best option.  We have a guest wifi network, but unfortunately, I don't think that any of our 5070s have the wifi components--I need to double-check that.  If it did then it solves the problem of not just installing but then updating it as well.  The only problem there is that technically I would be making a corporate LAN device available to a guest network device via the corporate device's second NIC going to the vidprod switch to the LAN port on the Wyse while the Wyse was also on the guest network.  That may get prickly.
